src/nacl/bindings/crypto_generichash.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2013 Donald Stufft and individual contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import, division, print_function
+
+from nacl import exceptions as exc
+
+from nacl._sodium import ffi, lib
+from nacl.utils import ensure
+
+crypto_generichash_BYTES = lib.crypto_generichash_blake2b_bytes()
+crypto_generichash_BYTES_MIN = lib.crypto_generichash_blake2b_bytes_min()
+crypto_generichash_BYTES_MAX = lib.crypto_generichash_blake2b_bytes_max()
+crypto_generichash_KEYBYTES = lib.crypto_generichash_blake2b_keybytes()
+crypto_generichash_KEYBYTES_MIN = lib.crypto_generichash_blake2b_keybytes_min()
+crypto_generichash_KEYBYTES_MAX = lib.crypto_generichash_blake2b_keybytes_max()
+crypto_generichash_SALTBYTES = lib.crypto_generichash_blake2b_saltbytes()
+crypto_generichash_PERSONALBYTES = \
+    lib.crypto_generichash_blake2b_personalbytes()
+
+_OVERLONG = '{0} length greater than {1} bytes'
+_TOOBIG = '{0} greater than {1}'
+
+
+def generichash_blake2b_salt_personal(data,
+                                      digest_size=crypto_generichash_BYTES,
+                                      key=b'', salt=b'', person=b''):
+    """One time hash interface"""
```

Comment:
might be nice if the docstring said "one *shot* hash interface", instead of one-time